File: nova-api/src/main/java/xyz/xenondevs/nova/api/data/NamespacedId.java
package xyz.xenondevs.nova.api.data;

import org.bukkit.NamespacedKey;
import org.jetbrains.annotations.NotNull;

public interface NamespacedId {
    
    /**
     * Gets the namespace of this {@link NamespacedId}.
     *
     * @return The namespace of this{@link NamespacedId}.
     */
    @NotNull String getNamespace();
    
    /**
     * Gets the name of this {@link NamespacedId}.
     *
     * @return The name of this {@link NamespacedId}.
     */
    @NotNull String getName();
    
    /**
     * Creates a {@link NamespacedKey} with the namespace and name of this {@link NamespacedId}.
     *
     * @return The {@link NamespacedKey}.
     */
    @NotNull NamespacedKey toNamespacedKey();
    
}

File: nova-api/src/main/kotlin/xyz/xenondevs/nova/api/data/NamespacedId.kt
package xyz.xenondevs.nova.api.data

import org.bukkit.NamespacedKey

interface NamespacedId {
    
    /**
     * The namespace of this [NamespacedId]
     */
    val namespace: String
    
    /**
     * The name of this [NamespacedId]
     */
    val name: String
    
    /**
     * Creates a [NamespacedKey] with the [namespace] and [name] of this [NamespacedId]
     */
    fun toNamespacedKey(): NamespacedKey
    
}